addons/l10n_cl/models/account_move.py

```diff
@@ -27,20 +27,23 @@ def _get_l10n_latam_documents_domain(self):
         domain = super()._get_l10n_latam_documents_domain()
         if (self.journal_id.l10n_latam_use_documents and
                 self.journal_id.company_id.country_id == self.env.ref('base.cl')):
+            domain += [('active', '=', True)]
             if self.type in ['in_invoice', 'in_refund']:
                 if self.partner_id.l10n_cl_sii_taxpayer_type == '2':
                     domain += [('code', '=', '71')]
                 return domain
             document_type_ids = self.journal_id.l10n_cl_sequence_ids.mapped('l10n_latam_document_type_id').ids
             domain += [('id', 'in', document_type_ids)]
             if self.partner_id.l10n_cl_sii_taxpayer_type == '3':
-                domain += [('code', 'in', ['35', '38', '39', '41'])]
+                domain += [('code', 'in', {'35', '38', '39', '41'})]
         return domain
 
     @api.constrains('type', 'l10n_latam_document_type_id')
     def _check_invoice_type_document_type(self):
         super()._check_invoice_type_document_type()
         for rec in self.filtered('l10n_latam_document_type_id'):
+            if rec.company_id.country_id != self.env.ref('base.cl'):
+                continue
```

Comment:
2 times filtering, maybe once.

Reply:
Improved in https://github.com/odoo/odoo/pull/37038/commits/712aeab2bd59a28cfb8914be85fab4ab6ecfdd1e